src/worker.cpp

```diff
@@ -486,11 +486,43 @@ void Worker::setDeviceTreeAndJson()
 
     if (fitConfigVal.find(devTreeFromJson) != std::string::npos)
     { // fitconfig is updated and correct JSON is set.
-
-        if (isSystemVPDOnDBus())
+        try
+        {
+            std::string l_backupRestoreJsonPath = getBackupRestoreJsonPath();
+            m_backupRestoreJson = utils::getParsedJson(l_backupRestoreJsonPath);
+            if (!m_backupRestoreJson.empty())
+            {
+                std::string l_dstLocation;
+                std::string l_dstPath;
+                getBackupRestorePathInfo("destination", l_dstLocation,
+                                         l_dstPath);
+
+                const std::string& l_invPath =
+                    m_parsedJson["frus"][SYSTEM_VPD_FILE_PATH].at(0).value(
+                        "inventoryPath", "");
+
+                if (isSystemVPDOnDBus() &&
```

Comment:
If system VPD presence on Dbus is primary check required to perform backup and restore, why do we need to postpone it till this point and do things which may be irrelevant after this check?